src/base/worker_pool.h

```diff
@@ -10,18 +10,27 @@ namespace base {
 
 class WorkerPool {
  public:
-  using NetWorker = Fn<void(const Atomic<bool>&, Data&)>;
-  using SimpleWorker = Fn<void(const Atomic<bool>&)>;
+  using NetWorker = Fn<void(WorkerPool*, Data&)>;
```

Comment:
Should be at least `const WorkerPool&` - since pool can't be modified inside worker, and pool can't be `nullptr`